### ISSUE TITLE
Rollbar Is Not Reporting Crashes Inside Android NativeModules

### ISSUE THREAD

**Issue by sdandois** [rollbar/rollbar-react-native#104]
I am using `"react-native": 0.59.8` and `"rollbar-react-native": 0.7.0`.

This is my code in `MainApplication.java`:

```java
  @Override
  public void onCreate() {
    super.onCreate();
    SoLoader.init(this, /* native exopackage */ false);
    RollbarReactNative.init(this, BuildConfig.ROLLBAR_POST_CLIENT_ACCESS_KEY, "environment");

  }
```

Also, there is an  onCreate method in `MainActivity.java`:

```java
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        SplashScreen.show(this);
        super.onCreate(savedInstanceState);
    }
```

Js only errros are being reported correctly.
Also, if I modify `onCreate` like this: 

```java
  @Override
  public void onCreate() {
    super.onCreate();
    SoLoader.init(this, /* native exopackage */ false);
    RollbarReactNative.init(this, BuildConfig.ROLLBAR_POST_CLIENT_ACCESS_KEY, "environment");

  throw new RuntimeException("error");

  }
```

It reports the error correctly.

But if a native module fails, there is no error. (I did this by modifying a installed module called DeviceInfo).

Help is much appreciated.

**Comment by waltjones** [comment]
@sdandois Thank you for the report. rollbar-java is used for the unhandled exception capture, and I'm seeing at least two issues.

Currently it's using `Thread.setUncaughtExceptionHandler()` to set the handler. This only receives exceptions for the thread it is called on. (There isn't a public interface to add the handler to additional threads.) This will be a problem for any React Native Modules. To capture for all threads, `Thread.setDefaultUncaughtExceptionHandler()` should be used. 
https://github.com/rollbar/rollbar-java/blob/master/rollbar-java/src/main/java/com/rollbar/notifier/Rollbar.java#L96

I'm also seeing that the BufferedSender gets set up with `ConcurrentLinkedQueue` instead of `DiskQueue`. (`DiskQueue` is only added if `config.sender()` is null, but rollbar-react-native already called `.build()` on the config builder so the default queue was already added.) `DiskQueue` is needed because in the Android environment, uncaught exceptions usually need to be sent after the app is restarted. This won't work with `ConcurrentLinkedQueue`.
https://github.com/rollbar/rollbar-java/blob/master/rollbar-android/src/main/java/com/rollbar/android/Rollbar.java#L275
https://github.com/rollbar/rollbar-react-native/blob/master/android/src/main/java/com/rollbar/RollbarReactNative.java#L47

There seems to be at least one more issue. I'll update here when I know more.

**Comment by waltjones** [comment]
Fix released in v0.8.0. https://github.com/rollbar/rollbar-react-native/releases/tag/v0.8.0
